src/serac/physics/utilities/functional/tuple.hpp

```diff
@@ -0,0 +1,408 @@
+#pragma once
+
+#include <utility>
+
+#include "serac/infrastructure/accelerator.hpp"
+
+namespace serac {
+
+template <typename... T>
```

Comment:
This file needs documentation and a comment explaining why we wrote our own tuple....

Reply:
```suggestion
// This is a class that mimics most of std::tuple's interface, except that it is usable in CUDA kernels and 
// admits some arithmetic operator overloads
template <typename... T>
```